junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/MethodSource.java

```diff
@@ -167,6 +175,67 @@ public boolean equals(Object o) {
 				&& Objects.equals(this.methodParameterTypes, that.methodParameterTypes);
 	}
 
+	/**
+	 * Get the {@linkplain Class Java class} of this source.
+	 *
+	 * <p>If the {@link Class} was not provided, but only the name, this method
+	 * attempts to lazily load the {@link Class} based on its name and throws a
+	 * {@link PreconditionViolationException} if the class cannot be loaded.
+	 *
+	 * @since 5.7
+	 * @see #getClassName()
+	 */
+	public final Class<?> getJavaClass() {
+		if (this.javaClass == null) {
+			this.javaClass = ReflectionUtils.tryToLoadClass(this.className).getOrThrow(
+				cause -> new PreconditionViolationException("Could not load class with name: " + this.className,
+					cause));
+		}
+		return this.javaClass;
+	}
+
+	/**
+	 * Get the {@linkplain Method Java method} of this source and any overloads.
+	 *
+	 * <p>This method attempts to lazily load the {@link Method} list based on
+	 * the method's name and throws a {@link PreconditionViolationException} if
+	 * the methods cannot be loaded.
+	 *
+	 * <p>An empty list will be returned if there are no matching methods.
+	 *
+	 * @since 5.7
+	 * @see #getMethodName()
+	 * @see #getJavaMethod()
+	 */
+	public List<Method> getJavaMethods() {
```

Comment:
As mentioned in https://github.com/junit-team/junit5/issues/2294#issuecomment-629152991, we are only adding `getJavaClass()` and `getJavaMethod()`.  Thus, please remove the `getJavaMethods()` method.

Reply:
sounds good